solidity/contracts/OngoingAirdrops.sol

```diff
@@ -111,7 +111,49 @@ contract OngoingAirdrops is AccessControl, IOngoingAirdrops {
     TokenAmount[] calldata _tokensAmounts,
     address _recipient,
     bytes32[] calldata _proof
-  ) internal virtual {}
+  ) internal virtual {
+    // Basic checks
+    if (_campaign == bytes32(0)) revert InvalidCampaign();
+    if (_claimee == address(0) || _recipient == address(0)) revert ZeroAddress();
+    if (_tokensAmounts.length == 0) revert InvalidTokenAmount();
+    if (_proof.length == 0) revert InvalidProof();
+
+    // Go through every token being claimed and apply check-effects-interaction per token.
+    bool _alreadyClaimed = true;
+    uint256[] memory _claimed = new uint256[](_tokensAmounts.length);
+    for (uint256 i = 0; i < _tokensAmounts.length; ) {
+      // Move calldata to memory
+      TokenAmount memory _tokenAmount = _tokensAmounts[i];
+      // Build our unique ID for campaign, token and user address.
+      bytes32 _campaignTokenAndUserId = _getIdOfCampaignTokenAndUser(_campaign, _tokenAmount.token, _claimee);
+      // Calculate to claim
+      _claimed[i] = _tokenAmount.amount - amountClaimedByCampaignTokenAndClaimee[_campaignTokenAndUserId];
+      // It might happen that not all airdropped tokens were updated.
+      if (_claimed[i] > 0) {
```

Comment:
We are accessing the array a lot. Could you check if it would be cheaper to store claimed in a mem variable and read it from there?

Reply:
Stack too deep 🌚  🌝

Reply:
@0xsambugs it didn't end up optimizing. Same costs.